platforms/ide/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/TestOperationMapper.java

```diff
@@ -103,12 +106,31 @@ private DefaultTestDescriptor toTestDescriptorForTest(OperationIdentifier buildO
         String operationDisplayName = test.toString();
 
         TestDescriptor originalDescriptor = getOriginalDescriptor(test);
-        if (!(originalDescriptor instanceof AbstractTestDescriptor)) {
+        if (originalDescriptor instanceof AbstractTestDescriptor) {
+            operationDisplayName = adjustOperationDisplayNameForIntelliJ(operationDisplayName, (AbstractTestDescriptor) originalDescriptor);
+        } else {
             operationDisplayName = getLegacyOperationDisplayName(operationDisplayName, originalDescriptor);
         }
         return new DefaultTestDescriptor(buildOperationId, test.getName(), operationDisplayName, test.getDisplayName(), InternalJvmTestDescriptor.KIND_ATOMIC, null, test.getClassName(), test.getName(), parentId, taskTracker.getTaskPath(buildOperationId));
     }
 
+    /**
+     * This is a workaround to preserve backward compatibility with IntelliJ IDEA.
+     * The problem only occurs in IntelliJ IDEA because it parses {@link OperationDescriptor#getDisplayName()} to get test display name.
+     * Once it's code is updated to use {@link org.gradle.tooling.events.test.TestOperationDescriptor#getTestDisplayName()}, the workaround can be removed as well.
```

Comment:
```suggestion
     * The problem only occurs in IntelliJ IDEA because it parses {@link OperationDescriptor#getDisplayName()} to get the test display name.
     * Once its code is updated to use {@link org.gradle.tooling.events.test.TestOperationDescriptor#getTestDisplayName()}, the workaround can be removed as well.
```